src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +452,317 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+                        editorCommandHandlerService.Execute((view, buffer) => new InvokeSignatureHelpCommandArgs(view, buffer), nextCommandHandler: null);
+
+                        return true;
+                    }
+                    else
+                    {
+                        _state.Clear(forceClearSymbolInformation: true);
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Creates a snippet for providing arguments to a call.
+        /// </summary>
+        /// <param name="methodName">The name of the method as it should appear in code.</param>
+        /// <param name="includeMethod"><see langword="true"/> to include the method name and invocation parentheses in
+        /// the resulting snippet; otherwise, <see langword="false"/> if the method name and parentheses are assumed to
+        /// already exist and the template will only specify the argument placeholders.</param>
+        /// <param name="parameters">The parameters to the method. If the specific target of the invocation is not
+        /// known, an empty array may be passed to create a template with a placeholder where arguments will eventually
+        /// go.</param>
+        /// <returns></returns>
+        private static XDocument CreateSnippet(string methodName, bool includeMethod, ImmutableArray<IParameterSymbol> parameters, CancellationToken cancellationToken)
+        {
+            XNamespace snippetNamespace = "http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet";
+
+            var template = new StringBuilder();
+
+            if (includeMethod)
+            {
+                template.Append(methodName).Append('(');
+            }
+
+            var declarations = new List<XElement>();
+            foreach (var parameter in parameters)
+            {
+                if (declarations.Any())
+                {
+                    template.Append(", ");
+                }
+
+                // Create a snippet field for the argument. The name of the field matches the parameter name, and the
+                // default value for the field is provided by a call to the internal ArgumentValue snippet function. The
+                // parameter to the snippet function is a serialized SymbolKey which can be mapped back to the
+                // IParameterSymbol.
+                template.Append('$').Append(parameter.Name).Append('$');
+                declarations.Add(new XElement(
+                    snippetNamespace + "Literal",
+                    new XAttribute(snippetNamespace + "Editable", "true"),
+                    new XElement(snippetNamespace + "ID", new XText(parameter.Name)),
+                    new XElement(snippetNamespace + "Function", new XText($"ArgumentValue({SymbolKey.CreateString(parameter, cancellationToken)})"))));
```

Comment:
i think this gives us a *minor* dependency that the symbolKey ToString doesn't produce any chars that would make it hard to parse ArgumentValue.  can you add any asserts that might be appropriate that the format of the string is ok?

Reply:
Since there is only one argument and the implementation includes everything between the first `(` and the last `)`, there are no restrictions:

https://github.com/dotnet/roslyn/blob/f5f54f773cedf07e5c646429be9adad520b401e3/src/EditorFeatures/Core.Cocoa/Snippets/AbstractSnippetExpansionClient.cs#L372-L374